### PR TITLE
Sve: Early versions of Intrinsics/Arm/Sve and hwintrinsiclistarm64sve

### PR DESCRIPTION
**Pull request by a74nh**
These files correspond to the latest vesion of the un-reviewed SVE API. They have all been auto generated.

hwintrinsiclistarm64sve.h cannot be fully autogenerated and will require manual edits.

### CODE REVIEW

**Comment by a74nh** [comment]
None of this code can be merged until the API is approved, and will require regenerating as it is changed.

Changes required for hwintrinsiclistarm64sve.h:

- `SIMD size` is -1 for everything. In once sense that makes sense because the vector size is unknown, but that isn't of much use. Maybe we could add a new flag `HW_Flag_element_size` which means that `SIMD size` is the vector element size.
- Some of the entries have multiple instructions. Eg `INS_SVE_CMPGT/INS_SVE_CMPLT`. The script does not have enough information to decide which one to use. This will need manually fixing.
- Some of the `HW_Flag_` flags need adding manually later, eg `HW_Flag_Commutative`.

My current thought is that when ready, this could be committed with _every line commented out_. Future patches would then uncomment and fix each line.

We also probably want to only do `SVE` and `SVE2` for now, ignoring all the other features.

**Comment by a74nh** [comment]
@kunalspathak, @tannergooding, @TamarChristinaArm, @BruceForstall

**Review by kunalspathak — COMMENTED** [review]
This looks great. 

> My current thought is that when ready, this could be committed with every line commented out. Future patches would then uncomment and fix each line.

Sounds good to me.

> Maybe we could add a new flag HW_Flag_element_size which means that SIMD size is the vector element size.

I am not sure if I follow the purpose of `HW_Flag_element_size`. They are and will be different depending on if they are `byte`, `short`, etc. 

> We also probably want to only do SVE and SVE2 for now, ignoring all the other features.

Can you tag all the issues that covers the API that you added in this PR?

Just a thought for future - If we decide to use your tool for future APIs as well, we have an option to just generate those new subset and copy/paste in this file OR, have your tool add just new entries and retain the existing entries (that were manually editted). That's how I did it for my Arm64Encoding tool. BTW, are you planning to open source your tool that generated the APIs as well as `hwintrinsiclistarm64sve.h`?

File: src/coreclr/jit/hwintrinsiclistarm64sve.h [hunk 1]
```diff
@@ -0,0 +1,684 @@
...
+HARDWARE_INTRINSIC(Sve,           CompareLessThanOrEqual,                                           -1,      2,      true,  {INS_SVE_CMPGE/INS_SVE_CMPLE,INS_SVE_CMPHS/INS_SVE_CMPLS,INS_SVE_CMPGE/INS_SVE_CMPLE,INS_SVE_CMPHS/INS_SVE_CMPLS,INS_SVE_CMPGE/INS_SVE_CMPLE,INS_SVE_CMPHS/INS_SVE_CMPLS,INS_SVE_CMPGE,      INS_SVE_CMPHS,      INS_SVE_FCMGE,      INS_SVE_FCMGE}    HW_Category_SIMD,                  HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen)
+HARDWARE_INTRINSIC(Sve,           CompareNotEqualTo,                                                -1,      2,      true,  {INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_CMPNE,      INS_SVE_FCMNE,      INS_SVE_FCMNE}    HW_Category_SIMD,                  HW_Flag_BaseTypeFromFirstArg|HW_Flag_SpecialCodeGen)
```
[the remark this answers is not in the document]
Comment (SingleAccretion) [reply src/coreclr/jit/hwintrinsiclistarm64sve.h:41]: > `EncodesExtraTypeArg`

This one should be assumed to be needed unless proven otherwise.

**Comment by a74nh** [comment]
> > Maybe we could add a new flag HW_Flag_element_size which means that SIMD size is the vector element size.
> 
> I am not sure if I follow the purpose of `HW_Flag_element_size`. They are and will be different depending on if they are `byte`, `short`, etc.

Yes, your right. It just should be -1 throughout.

> 
> > We also probably want to only do SVE and SVE2 for now, ignoring all the other features.
> 
> Can you tag all the issues that covers the API that you added in this PR?

Sure. This is everything in https://github.com/dotnet/runtime/issues/93095#issuecomment-1778932195

Which is...

FEAT_SVE:
- [ ] https://github.com/dotnet/runtime/issues/93887 
- [ ] https://github.com/dotnet/runtime/issues/94003
- [ ] https://github.com/dotnet/runtime/issues/94004
- [ ] https://github.com/dotnet/runtime/issues/94005
- [ ] https://github.com/dotnet/runtime/issues/94006
- [ ] https://github.com/dotnet/runtime/issues/94007
- [ ] https://github.com/dotnet/runtime/issues/94008
- [ ] https://github.com/dotnet/runtime/issues/93964
- [ ] https://github.com/dotnet/runtime/issues/94009
- [ ] https://github.com/dotnet/runtime/issues/94011
- [ ] https://github.com/dotnet/runtime/issues/94014

FEAT_SVE2:
- [ ] https://github.com/dotnet/runtime/issues/94015
- [ ] https://github.com/dotnet/runtime/issues/94017
- [ ] https://github.com/dotnet/runtime/issues/94018
- [ ] https://github.com/dotnet/runtime/issues/94019
- [ ] https://github.com/dotnet/runtime/issues/94020
- [ ] https://github.com/dotnet/runtime/issues/94021
- [ ] https://github.com/dotnet/runtime/issues/94022
- [ ] https://github.com/dotnet/runtime/issues/94023

Other SVE Features:
- [ ] https://github.com/dotnet/runtime/issues/94026 (8.2, mandatory if SVE is available)
- [ ] https://github.com/dotnet/runtime/issues/94027 (8.2, optional. From 8.6 is mandatory if SVE is available)
- [ ] https://github.com/dotnet/runtime/issues/94028 (8.2, optional. From 8.6 is mandatory if SVE is available)
- [ ] https://github.com/dotnet/runtime/issues/94024 (8.2, optional. Not in any available hardware)
- [ ] https://github.com/dotnet/runtime/issues/94025 (8.2, optional. Not in any available hardware)
- [ ] https://github.com/dotnet/runtime/issues/94423 (9.0, optional)
- [ ] https://github.com/dotnet/runtime/issues/94424 (9.0, optional)
- [ ] https://github.com/dotnet/runtime/issues/94425 (9.0, optional)
- [ ] https://github.com/dotnet/runtime/issues/94426 (9.0, optional)


> Just a thought for future - If we decide to use your tool for future APIs as well, we have an option to just generate those new subset and copy/paste in this file OR, have your tool add just new entries and retain the existing entries (that were manually editted). That's how I did it for my Arm64Encoding tool. 

Yes, I could fix up the tool to read in the existing files and fill in the gaps.

>BTW, are you planning to open source your tool that generated the APIs as well as `hwintrinsiclistarm64sve.h`?

Happy to open source it. It's in python. And it's a little messy as there is a lot of renaming and fixing up.

**Pull request closed without merging**